airbyte-cdk/bulk/toolkits/load-db/src/main/kotlin/io/airbyte/cdk/load/orchestration/db/legacy_typing_deduping/TableNameMapFactory.kt

```diff
@@ -32,23 +38,185 @@ class TableCatalogFactory(
 ) {
     @Singleton
     fun get(): TableCatalog {
-        // TODO handle collisions in table names
-        return TableCatalog(
-            catalog.streams.associateWith { stream ->
+        if (catalog.streams.isEmpty()) {
+            throw ConfigErrorException(
+                "The catalog contained no streams. This likely indicates a platform/configuration error."
```

Comment:
fyi: bulk CDK handles this already, we can remove the check here 

(https://github.com/airbytehq/airbyte/blob/b88b37bed747ff7b23b627197be1495cde059fe4/airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/command/DestinationCatalog.kt#L26)